NitroxClient/Communication/Packets/Processors/VehicleNameChangeProcessor.cs

```diff
@@ -18,12 +19,36 @@ public VehicleNameChangeProcessor(IPacketSender packetSender)
 
         public override void Process(VehicleNameChange namePacket)
         {
-            GameObject target = NitroxEntity.RequireObjectFrom(namePacket.Id);
-            SubNameInput subNameInput = target.RequireComponentInChildren<SubNameInput>();
+            GameObject vehicleObject = NitroxEntity.RequireObjectFrom(namePacket.VehicleId);
+            SubNameInput subNameInput = null;
+            SubName subNameTarget = null;
 
-            using (packetSender.Suppress<VehicleNameChange>())
+            if (vehicleObject.GetComponent<Vehicle>())
             {
-                subNameInput.OnNameChange(namePacket.Name);
+                if (namePacket.ParentId.HasValue)
+                {
+                    GameObject moonpool = NitroxEntity.RequireObjectFrom(namePacket.ParentId.Value);
+                    GameObject baseCell = moonpool.RequireComponentInParent<BaseCell>().gameObject;
+
+                    subNameInput = baseCell.RequireComponentInChildren<SubNameInput>();
+                    subNameTarget = vehicleObject.RequireComponentInChildren<SubName>();
+                }
+            }
+            else
+            {
+                subNameInput = vehicleObject.RequireComponentInChildren<SubNameInput>();
+                subNameTarget = subNameInput.target;
+            }
+
+            if (subNameInput != null && subNameTarget != null)
```

Comment:
```suggestion
            if (subNameInput && subNameTarget)
```
Also you should add a error log here so we can now when it's failing